weave/type_handlers/DateTime/datetime.py

```diff
@@ -0,0 +1,59 @@
+"""Defines the custom DateTime weave type."""
+
+from __future__ import annotations
+
+import datetime
+import json
+from typing import TextIO, cast
+
+from weave.trace import serializer
+from weave.trace.custom_objs import MemTraceFilesArtifact
+
+
+class DatetimeWithArtifact(datetime.datetime):
```

Comment:
can you help me understand why we need this? Seems like a fault of some internal problem?